docs/deploying-applications/azure-deployments/ase/index.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Azure App Service Environments
+description: This guide covers deploying to Azure App Service Environments.
+---
+
+This guide covers deploying apps in to Azure [App Service Environments](https://docs.microsoft.com/en-au/azure/app-service/environment/intro). It does not cover how to deploy/setup an ASE itself.
+
+From an Octopus user perspective, deploying to an ASE is usually no different to deploying to any other app service in Azure. While the app services within an ASE are isolated, the management interface for managing them and deploying to them is the same as for any other app service. Usually.
+
+## Internal App Service Environments
+
+Internal ASEs are where the wheels usually come off for deploying from Octopus. An [internal ASE](https://docs.microsoft.com/en-us/azure/app-service/environment/create-ilb-ase) is one that has an Internal Load Balancer (ILB). By definition it therefore cannot be accessed from the internet, it's designed to host internal apps (i.e. intranet apps).
+
+Given that you can't access the app, or its management endpoint (Kudu), from the internet you therefore can't deploy to it from Octopus without some network setup. To help explain what's required let's look at what happens when you deploy to a web app in Azure from Octopus.
+
+1. Octopus server creates a deployment task
+2. Task scheduler picks up the task and hands the work over to Calamari
+3. Calamari picks up all the information about the deployment and connects to Azure
+4. Calamari locates the resource group that's being deployed to (there's a reason we do it this way, see [below](#resource_groups))
+5. Calamari locates the web app within the resource group and requests it's publish profile from Azure
+6. Calamari then hands over to the [DeploymentManager](https://msdn.microsoft.com/en-us/library/microsoft.web.deployment.deploymentmanager(v=vs.90).aspx).[SyncTo](https://msdn.microsoft.com/en-us/library/dd543271(v=vs.90).aspx) method to actually do the deployment
+
+Contained in the publish profile is the Uri of the deployment endpoint (Kudu) for the web app. This is the critical piece here.
+
+For an external ASE that Uri will be publicly accessible (e.g. https://yourapp.scm.aseName.p.azurewebsites.net).
+
+For an internal ASE the Uri will not be publicly accessible, it will be something like https://yourapp.scm.yourdomain  This is where the deployments will fail, they will be able to see all of the other Urls required but when they get to step 6 Octopus won't be able to resolve the address for the Uri.
+
+To fix that you need 2 things to happen. First, the network the Octopus server is on has to be connected to the ASE's VNet, e.g. using ExpressRoute or a VPN. Second, the Octopus server needs to be able to resolve `yourapp.scm.yourdomain` to the IP address of your Azure ILB (found in the **IP addresses** for the ASE in the Azure portal), e.g. through DNS configuration.
```

Comment:
> to the IP address of your Azure ILB

Could we specify `Internal Load Balancer IP Address` instead of `IP Address` as there is also the management and outbound IP which might get confusing?

Reply:
I've added in this changes.